Add a new ESP32 development board.

HalfSweet opened the pull request:
An AirM2M Core ESP32S3 development board was added to the ESP32 port, here is its official website: <https://wiki.luatos.com/chips/esp32s3/index.html>
This time two firmware were added, one using the included USB to serial chip (CH343) and the other using the USB-Serial/JTAG in the ESP32S3 chip. on the development board, we used a switch to switch.
After testing, they work fine.

HalfSweet commented:
I'm very sorry, but I don't know how I should change the format of such a commit.

robert-hh commented:
The E-Mail Address is not acceptable. You used `60973476+HalfSweet@users.noreply.github.com`
But you have to use an E-Mail address directly linked to you. See: https://github.com/micropython/micropython/blob/master/CODECONVENTIONS.md
As far as I understand it, you may as well sign the commit. As a minor note, the commit message format is wrong. It could be something like:
`esp32/boards: Add the AirM2M  Core ESP32S3 development board.`
For the options of using UART or USB, you might go for variants instead of having two board definitions.

HalfSweet committed:
ports/esp32/board/AirM2M_CORE_ESP32S3_USB: Add new board.

esp32/boards: More info: <https://wiki.luatos.com/chips/esp32s3/index.html>.

Signed-off-by: HalfSweet <HalfSweet@HalfSweet.cn>

HalfSweet committed:
ports/esp32/board/AirM2M_CORE_ESP32S3_UART: Add new board.

esp32/boards: Add the AirM2M Core ESP32S3 development board, More info: <https://wiki.luatos.com/chips/esp32s3/index.html>.

Signed-off-by: HalfSweet <HalfSweet@HalfSweet.cn>
